### PR TITLE
Remove deprecated createParser methods

### PR DESCRIPTION
**Pull request by dakrone**
This removes the final instances of the callers of `XContent.createParser` and
`XContentHelper.createParser` that did not pass in the `DeprecationHandler`. It
also removes the now-unused deprecated methods and fully removes any mention of
Log4j or LoggingDeprecationHandler from the XContent code.

Relates to #28504

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/xcontent/XContent.java [hunk 1]
```diff
@@ -115,59 +115,4 @@ XContentParser createParser(NamedXContentRegistry xContentRegistry,
...
-    /**
-     * Creates a parser over the provided reader using
-     * {@link LoggingDeprecationHandler}.
-     * @deprecated This is a temporary shim so we can migrate all calls to createParser incrementally.
-     *      Use {@link #createParser(NamedXContentRegistry, DeprecationHandler, Reader)} instead.
-     */
-    @Deprecated
-    default XContentParser createParser(NamedXContentRegistry xContentRegistry, Reader reader) throws IOException {
-        return createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, reader);
-    }
```
Comment (imotov) [inline server/src/main/java/org/elasticsearch/common/xcontent/XContent.java]: ❤️

File: server/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java [hunk 2]
```diff
@@ -316,7 +316,7 @@ public void writeRawField(String name, InputStream content, XContentType content
         if (mayWriteRawData(contentType) == false) {
             // EMPTY is safe here because we never call namedObject when writing raw data
             try (XContentParser parser = XContentFactory.xContent(contentType)
-                    .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, content)) {
+                    .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, content)) {
```
Comment (imotov) [inline server/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java]: I think an explanation why it's ok to throw an exception here might be helpful for future us.

**Review by imotov — APPROVED** [review]
LGTM! Nice clean up!

File: server/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java (hunk 2 above)
Comment (dakrone) [reply server/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java]: Sure, I'll add a comment, good idea

**Pull request merged**